File: backend/src/main/java/com/example/demo/dto/Customer.java
package com.example.demo.dto;

import com.example.demo.entity.Address;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Customer {
    private String firstName;

    private String lastName;

    private String email;

    private Address shippingAddress;

    private Address billingAddress;
}

File: frontend/src/app/models/customer.ts
import { OrderPlaced } from './orderPlaced';

export type Customer = {
  firstName: string;
  lastName: string;
  email?: string;
  shippingAddress?: Address;
  billingAddress?: Address;
  orders?: OrderPlaced[];
  _links?: {
    self: {
      href: string;
    };
    customer: {
      href: string;
    };
  };
};

export type Address = {
  firstName: string;
  lastName: string;
  street: string;
  city: string;
  state: string;
  zip: string;
  _links?: {
    customer: {
      href: string;
    };
  };
};
